core/templates/dev/head/filters/string-utility-filters/format-timer.pipe.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview FormatTimer filter for Oppia.
+ */
+
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({name: 'formatTimer'})
+export class FormatTimerPipe implements PipeTransform {
+  transform(input: any): string {
```

Comment:
Shouldn't input be a number?

Reply:
@ankita240796 it can be a string also, and I believe the pipes should accept any data type...
@bansalnitish should I change this to number?

Reply:
It should be `number`.

Reply:
Hey Guys, @ankita240796 @bansalnitish , I tried changing this to number but in our service
```      let transformedDate = this.formatTimePipe.transform(date);``` we are passing a date to this filter so a number is not a correct data type for this and since we are doing arithmetic operations in the filter, it's supposed to be a number. So i dont know what to do, maybe add a Todo for now. Or if you guys have some suggestion, then please let me know.

Reply:
Update the service caller to convert the date to a number first before passing it to this filter. (In general, sometimes, when enforcing types, you'll need to fix cases of non-compliance.)

Reply:
Ok done!